pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -48,39 +48,44 @@ func (d *DualWriterMode2) Mode() DualWriterMode {
 }
 
 // Create overrides the behavior of the generic DualWriter and writes to LegacyStorage and Storage.
-func (d *DualWriterMode2) Create(ctx context.Context, original runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
+func (d *DualWriterMode2) Create(ctx context.Context, in runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
 	var method = "create"
 	log := d.Log.WithValues("method", method)
 	ctx = klog.NewContext(ctx, log)
 
 	startLegacy := time.Now()
-	created, err := d.Legacy.Create(ctx, original, createValidation, options)
+	createdFromLegacy, err := d.Legacy.Create(ctx, in, createValidation, options)
 	if err != nil {
 		log.Error(err, "unable to create object in legacy storage")
 		d.recordLegacyDuration(true, mode2Str, d.resource, method, startLegacy)
-		return created, err
+		return createdFromLegacy, err
 	}
 	d.recordLegacyDuration(false, mode2Str, d.resource, method, startLegacy)
 
-	if err := enrichLegacyObject(original, created); err != nil {
-		return created, err
+	accIn, err := meta.Accessor(in)
+	if err != nil {
+		return createdFromLegacy, err
+	}
+	if accIn.GetUID() != "" {
+		return nil, fmt.Errorf("there is an UID and it should not: %v", accIn.GetUID())
 	}
 
 	startStorage := time.Now()
-	rsp, err := d.Storage.Create(ctx, created, createValidation, options)
+	createdFromStorage, err := d.Storage.Create(ctx, in, createValidation, options)
```

Comment:
I am still skeptical that we are not using `createdFromLegacy` as the input to this second Create.

At the very least, we need to ensure that the names returned from both are the same.  and fail if not?